vulyk/models/task_types.py

```diff
@@ -63,78 +77,201 @@ def export_reports(self, qs=None):
             yield task.get_results()
 
     def get_next(self, user):
-        """Finds given user a new task
-        Assumes that user is eligible for this kind of tasks
+        """
+        Finds given user a new task and starts new WorkSession
 
-        Args:
-            user: an instance of User model
-        Returns:
-            Prepared dictionary or model, or None
+        :param user: an instance of User model
+        :type user: models.User
 
-        Raises:
-            TaskPermissionError
+        :returns: Prepared dictionary of model, or empty dictionary
+        :rtype: dict
         """
+        task = self._get_next_task(user)
 
-        res = self.task_model.objects(
+        if task is not None:
+            self._start_work_session(task, user)
+
+            return task.as_dict()
+        else:
+            return {}
+
+    def _get_next_task(self, user):
+        """
+        Finds given user a new task
+
+        :param user: an instance of User model
+        :type user: models.User
+
+        :returns: Model instance or None
+        :rtype: AbstractTask
+        """
+        task = None
+        rs = self.task_model.objects(
             task_type=self.type_name,
-            users_count__lt=self.redundancy,
             users_processed__ne=user.id,
-            users_skipped__ne=user.id)
-        res = res[random.randint(0, res.count())]
+            users_skipped__ne=user.id,
+            closed__ne=True)
+
+        if rs.count() > 0:
+            task = rs[random.randint(0, rs.count())]
```

Comment:
Notice: `rs[random.randint(0, rs.count())] ` returns particular task not a QuerySet.
Turnin' back to your Q: we still need the task to be returned due to `WorkSession` creation need.
Look at  line 92.